resources/ad_applications.go

```diff
@@ -0,0 +1,611 @@
+						Type:        schema.TypeJSON,
+					},
+					{
+						Name:        "resource_access",
+						Description: "The list of OAuth20 permission scopes and app roles that the application requires from the specified resource",
+						Type:        schema.TypeJSON,
+						Resolver:    resolveAdApplicationRequiredResourceAccessResourceAccess,
+					},
+					{
+						Name:        "resource_app_id",
+						Description: "The unique identifier for the resource that the application requires access to This should be equal to the appId declared on the target resource application",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("ResourceAppID"),
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchAdApplications(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	svc := meta.(*client.Client).Services().AD.Applications
+	response, err := svc.List(ctx, "")
+	if err != nil {
+		return err
+	}
+	for response.NotDone() {
+		res <- response.Values()
+		if err := response.NextWithContext(ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func resolveAdApplicationOptionalClaims(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	app, ok := resource.Item.(graphrbac.Application)
+	if !ok {
+		return fmt.Errorf("not a graphrbac.Application instance: %#v", resource.Item)
+	}
+	out, err := json.Marshal(app.OptionalClaims)
+	if err != nil {
+		return err
+	}
+	return resource.Set(c.Name, out)
+}
+
+func fetchAdApplicationAppRoles(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	app, ok := parent.Item.(graphrbac.Application)
+	if !ok {
+		return fmt.Errorf("not a graphrbac.Application instance: %#v", parent.Item)
+	}
+	if app.AppRoles != nil {
+		for _, item := range *app.AppRoles {
+			res <- item
+		}
+	}
+	return nil
+}
+
+func fetchAdApplicationKeyCredentials(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	app, ok := parent.Item.(graphrbac.Application)
+	if !ok {
+		return fmt.Errorf("not a graphrbac.Application instance: %#v", parent.Item)
+	}
+	if app.KeyCredentials != nil {
+		for _, item := range *app.KeyCredentials {
+			res <- item
+		}
+	}
+	return nil
+}
+
+func fetchAdApplicationOauth2Permissions(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	app, ok := parent.Item.(graphrbac.Application)
+	if !ok {
+		return fmt.Errorf("not a graphrbac.Application instance: %#v", parent.Item)
+	}
+	if app.Oauth2Permissions != nil {
```

Comment:
nesting makes it harder to read
please take a look at other similar items
```
	if app.Oauth2Permissions == nil {
		return nil
	}
	for _, item := range *app.Oauth2Permissions {
		res <- item
	}
```

Reply:
Fixed as suggested.